website/project/views/contributor.py

```diff
@@ -669,6 +670,12 @@ def claim_user_form(auth, **kwargs):
     user.update_guessed_names()
     # The email can be the original referrer email if no claimer email has been specified.
     claimer_email = unclaimed_record.get('claimer_email') or unclaimed_record.get('email')
+
+    # If there is a registered user with this email, redirect to 're-enter password' page
+    user_from_email = User.find_one(Q('username', 'eq', claimer_email))
```

Comment:
@brianjgeiger the current [dev branch](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/views/contributor.py#L674) looks different: 

In https://github.com/CenterForOpenScience/osf.io/commit/b885376a04cd40ad995716ea04536a06d1819350:

```python
# If there is a registered user with this email, redirect to 're-enter password' page
    user = User.find_by_email(claimer_email)
    user_from_email = user[0] if user else None
    if user_from_email and user_from_email.is_registered:
        return redirect(web_url_for('claim_user_registered', uid=uid, pid=pid, token=token))
```

In develop:
```python
 # If there is a registered user with this email, redirect to 're-enter password' page
    found_by_email = User.find_by_email(claimer_email)
    user_from_email = found_by_email[0] if found_by_email else None
    if user_from_email and user_from_email.is_registered:
        return redirect(web_url_for('claim_user_registered', uid=uid, pid=pid, token=token))
```